### ISSUE TITLE
Wrong request detection with leading whitespace & non http method

### ISSUE THREAD

**Issue by rngtng** [AnWeber/httpyac#447]
Hey hey

I'm quite surprised to find out that leading whitespace in front of the http method is supported. Can you elaborate why? IMHO it leads to bad side effects and allow writing (IMHO)  rather unreadable code.  Like this is a valid syntax:

```http
            GET https://httpbin.org/anything
```

One bad side effect is that then body is dropped in the following:

```http
POST https://webhook.site/....
Content-Type: text/plain
     Post the following content
```

And even more surprising, it fails with non http method too. This is rendered as three requests:

```http
POST https://webhook.site/....
PostCode is 1234
Gettogether
```

My expectation is that lines with leading whitespace are not detected as new requests. Same for lines which have start with HTTP method. wyt?

**Comment by AnWeber** [comment]
I actually don't use whitespaces before the HTTP method either. I would like to check how Intellij behaves here and act accordingly. I think I would adjust the regex here to only accept uppercase letters.

**Comment by rngtng** [comment]
Yes. Sounds like a good fix

**Comment by AnWeber** [comment]
So Intellij allows whitespaces, but the request method must be specified with uppercase. I adjust my logic accordingly to avoid false positives. Additionally I add a new switch `# @force-separator` to force the explicit use of `###` to terminate the current region.

**Comment by rngtng** [comment]
@AnWeber ok, I see. Thanks for Introducing `# @force-separator` although it feels a little bit like a hack and not very intuitive. How come `PostCode` is matched at all?  I guess the line in question is this one:
https://github.com/AnWeber/httpyac/blob/main/src/plugins/core/parse/requestHttpRegionParser.ts#L8 

I added it here https://regex101.com/r/XwdLRc/1

I'd assume a single space after the method needs to be enforced (to seperate method and url) so what about

`... RAPHQL)\s+(?<url>.+?)$`
or
`... RAPHQL) \s*(?<url>.+?)$`

IMHO this would sharpen the parsing and fix my cases?

**Comment by rngtng** [comment]
on top, a url can't have whitespaces? so you probably could sharpen the matching here too.. 

I quick found this reference for allowed URL characters: https://www.freecodecamp.org/news/url-encoded-characters-reference/#:~:text=There%20are%20only%20certain%20characters,that%20can%20have%20special%20meanings.

**Comment by AnWeber** [comment]
@rngtng Unfortunately more difficult. The url can also contain variables that maybe allow whitespace. Now I could check if there is a `{{` before the whitespace. But I lose the independence of the parser. And additionally I would limit the extensibility, because there might be use cases for whitespaces without `{{` (not encoded query params?).

**Comment by AnWeber** [comment]
I forgot to make the whitespace after the method mandatory. I will add `+`

**Comment by rngtng** [comment]
Ah right. Forgot about the variables. But imho enforcing a single whitespace would solve it already. With the leading one i can live with.

so imho just add the + and remove the force separator to keep it simple?

**Comment by AnWeber** [comment]
The force separator would just be a workaround in case I run into a problem again. But it would also be solvable by creative variables. I remove it again. The `+` is already in there. I just forgot to reference the ticket.

**Comment by AnWeber** [comment]
change is released with 6.4.3

**Comment by rngtng** [comment]
thanks a lot!
